### PR TITLE
Fix dir list streaming bug

### PR DESCRIPTION
**Pull request by chrislearn**
https://github.com/salvo-rs/salvo/issues/29#issuecomment-840434616

### CODE REVIEW

**Comment by biluohc** [comment]
感谢，

不过也许
```
pub use dir::{StaticDir, Options};
```
比 
```
pub mod dir;
pub mod fs;

pub use dir::StaticDir;
pub use fs::StaticFile;
```
好一些？

**Comment by chrislearn** [comment]
Option s 直接拉出来意义好像不那么明确

**Comment by biluohc** [comment]
主要是options这个名字吧，，模块都公开的话StaticDir/StaticFile就会有两个路径可用了，，虽然也不是什么比较大的问题

File: extra/src/compression/mod.rs [hunk 1]
```diff
@@ -71,23 +71,47 @@ impl Handler for CompressionHandler {
...
-                let stream = ReaderStream::new(BrotliEncoder::new(StreamReader::new(body)));
-                res.streaming(stream);
+        if let Some(body) = res.take_body() {
+            match body {
+                Body::Empty => {
+                    return;
+                }
+                Body::Bytes(body) => {
+                    let reader =
+                        StreamReader::new(tokio_stream::iter(vec![Result::<_, std::io::Error>::Ok(body)]));
```
Comment (driftluo) [inline extra/src/compression/mod.rs:81]: use `once` instead, don't need vec

**Review by driftluo — COMMENTED** [review]
I think you can test the compression ratio and set the lower limit of compression for users. Small data compression is an idling CPU, which has no practical meaning.

**Comment by biluohc** [comment]
压缩确实是，太小的body压缩后会比原始的还大

**Pull request merged**